lib/llvmopencl/ParallelRegion.cc

```diff
@@ -441,47 +441,61 @@ ParallelRegion::Verify()
   return true;
 }
 
+#ifdef LLVM_OLDER_THAN_8_0
+#define PARALLEL_MD_NAME "llvm.mem.parallel_loop_access"
+#else
+#define PARALLEL_MD_NAME "llvm.access.group"
+#endif
+
 /**
  * Adds metadata to all the memory instructions to denote
  * they originate from a parallel loop.
  *
  * Due to nested parallel loops, there can be multiple loop
  * references.
  *
- * Format:
- * llvm.mem.parallel_loop_access !0
+ * Format (LLVM 8+):
+ *
+ *     !llvm.access.group !0
  *
- * !0 { metadata !0 }
+ *     !0 distinct !{}
  *
  * In a 2-nested loop:
  *
- * llvm.mem.parallel_loop_access !0
+ *     !llvm.access.group !0
+ *
+ *     !0 { !1, !2 }
+ *     !1 distinct !{}
+ *     !2 distinct !{}
  *
- * !0 { metadata !1, metadata !2}
- * !1 { metadata !1 }
- * !2 { metadata !2 }
+ * Parallel loop metadata on memory reads also implies that
+ * if-conversion (i.e., speculative execution within a loop iteration)
+ * is safe. Given an instruction reading from memory,
+ * IsLoadUnconditionallySafe should return whether it is safe under
+ * (unconditional, unpredicated) speculative execution.
```

Comment:
Can you add a link to the LLVM bug here?

Reply:
ac4468e6cf68c882ed16b995d44d86df08b45ec9